sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetSchemaUtils.scala

```diff
@@ -568,9 +576,8 @@ object ParquetSchemaUtils extends Arm {
   private def evolveSchemaCasts(cv: ColumnView, dt: DataType): ColumnView = {
     if (needDecimalCast(cv, dt)) {
       cv.castTo(DecimalUtil.createCudfDecimal(dt.asInstanceOf[DecimalType]))
-    } else if (needUnsignedToSignedCast(cv, dt)) {
-      cv.castTo(DType.create(GpuColumnVector.getNonNestedRapidsType(dt).getTypeId))
-    } else if (needInt32Downcast(cv, dt)) {
+    } else if (needUnsignedToSignedCast(cv, dt) || needInt32Downcast(cv, dt) ||
+        needSignedUpcast(cv, dt)) {
```

Comment:
So there is no support for signed => unsigned cast in Spark, right? And there is also no support for any casting with (un)signed 64 bit integers? Can you add a link to Spark's docs to clarify that please?

Reply:
Spark doesn't have unsigned types.